### PR TITLE
parse-expr instead of parse-range in parse-do

### PR DESCRIPTION
**Pull request by innerlee**

### CODE REVIEW

**Comment by JeffBezanson** [comment]
What's the motivation for this?

Also will need a test case in test/syntax.jl.

**Comment by innerlee** [comment]
Oh silly of me. Feel free to close this!

File: test/syntax.jl [hunk 1]
```diff
@@ -1436,3 +1436,6 @@ let ex = Meta.parse("@test27521(2) do y; y; end")
                      fex)
     @test macroexpand(@__MODULE__, ex) == Expr(:tuple, fex, 2)
 end
+
+# pr #27668
+@test (try Meta.parse("map(1) do x...; end"); catch; 0; end) isa Expr
```
Comment (JeffBezanson) [inline test/syntax.jl]: There's no need for the try-catch, since an unexpected error will cause the test to fail anyway.

Comment (innerlee) [reply test/syntax.jl]: If it throw, will the exception block the tests after it?

Comment (JeffBezanson) [reply test/syntax.jl]: No; `@test` wraps its argument in try-catch and records whether an exception was thrown.

**Pull request closed without merging**